proposals/2023-08-21-utf8.md

```diff
@@ -0,0 +1,159 @@
+## UTF8 Support for Metric and Label names
+
+* **Owners:**
+  * `<@author: owen.williams@grafana.com>`
+
+* **Implementation Status:** `Proof-of-concept implemented`
+
+* **Related Issues and PRs:**
+  * [GH Issue](https://github.com/prometheus/prometheus/issues/12630)
+  * [PR](https://github.com/grafana/mimir-prometheus/pull/476) (TODO needs to be moved)
+
+* **Other docs or links:**
+  * [Background Discussion / Justification](https://docs.google.com/document/d/1yFj5QSd1AgCYecZ9EJ8f2t4OgF2KBZgJYVde-uzVEtI/edit).  Please read this document first for more information on the chosen solution.
+
+> TL;DR: We propose to add support for the UTF8 characters in Prometheus metric and label names. While tsdb and the Prometheus code already support arbitrary strings, PromQL and the exposition format need a quoting syntax to make these values distinguishable.
+
+## Why
+
+We wish to use Prometheus as a backing database for other metrics implementations like Open Telemetry that use Prometheus-illegal characters, like '.'. Establishing a quoting syntax avoids the need for verbose and unreadable character encoding schemes, as well as non-roundtripable replacement schemes like using underscore for all unsupported characters.
+
+### Pitfalls of the current solution
+
+The current recommendation is to replace all unsupported characters with underscores. This substitution cannot be reversed, creating possible name collisions. Furthermore, the change in metric names between original write and eventual read can be confusing for end users.
+
+## Goals
+
+* Allow arbitrarily-named metric and label names to be submitted to Prometheus.
+* Allow these metric and label names to be easily queried back with PromQL.
+* Ensure that Go templating supports quoted label names.
+* Provide a graceful backwards-compatible fallback for query clients that do not support UTF8 metric and label names in returned results.
+
+### Audience
+
+The audience for this change is very broad. It would allow all end-users of Prometheus to use richer metric names. Tool and service authors will want to be aware of the change and support the new character set.
+
+## Non-Goals
+
+We do not directly address the issue of adding function-like syntaxes, such as `metricname.sum`. The changes proposed here should be forward-compatible with such a new syntax, as long as the new syntax is outside of the quoted metric names.
+
+We do not propose a syntax for selecting metric names other than by basic equality. The changes proposeed here would be forward-compatible with ways of doing that, for instance operator prefixes on the quoted metric name.
+
+## How
+
+We propose the following change in syntax to PromQL and the exposition format:
+
+* Label names may be quoted
```

Comment:
Should this be more specific on what sort of quotes are supported? (double quotes, single quotes and backticks)